### DISCUSSION TITLE
are the template images used?

### DISCUSSION THREAD

**Discussion by ltzhuchen** [STVIR/pysot#284] [General]
From the dataset.py, I failed to find the loading of ***.z.jpg.
Both the template and search are obtained from:
 def get_image_anno(self, video, track, frame):
        frame = "{:06d}".format(frame)
        image_path = os.path.join(self.root, video,
                                  self.path_format.format(frame, track, 'x'))
        image_anno = self.labels[video][track][frame]
        return image_path, image_anno

I'm wondering whether the previous saved ***.z.jpg are used during training?

**Comment by xxAna** [comment]
I found the same problem. It seems that when the dataset.py read the image, it will crop it according to specified size in config. So the x image in dataset is 511,which is much larger than 255 and 287. So I think the z image can delete if the space is limited?

**Comment by liwenjielongren** [comment]
@ltzhuchen @xxAna 你们好，请问这个问题是什么原因呢，是z.jpg不需要保存，还是说调用template的路径是改为self.path_format.format(frame, track, 'z'))

**Comment by laisimiao** [comment]
@zyc-ai please give an answer(because you place a duplicate flag on #406)
